extensions/ql-vscode/src/view/variant-analysis/VariantAnalysisAnalyzedRepoItem.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+import { useCallback, useState } from 'react';
+import styled from 'styled-components';
+import { VSCodeBadge } from '@vscode/webview-ui-toolkit/react';
+import { isCompletedAnalysisRepoStatus, VariantAnalysisRepoStatus } from '../../remote-queries/shared/variant-analysis';
+import { formatDecimal } from '../../pure/number';
+import { Codicon, ErrorIcon, LoadingIcon, SuccessIcon } from '../common';
+import { Repository } from '../../remote-queries/shared/repository';
+import { AnalysisAlert, AnalysisRawResults } from '../../remote-queries/shared/analysis-result';
+import { AnalyzedRepoItemContent } from './AnalyzedRepoItemContent';
+
+// This will ensure that these icons have a className which we can use in the TitleContainer
+const ExpandCollapseCodicon = styled(Codicon)``;
+
+const TitleContainer = styled.button`
+  display: flex;
+  gap: 0.5em;
+  align-items: center;
+
+  color: var(--vscode-editor-foreground);
+  background-color: transparent;
+  border: none;
+  cursor: pointer;
+
+  &:disabled {
+    cursor: default;
+
+    ${ExpandCollapseCodicon} {
+      color: var(--vscode-disabledForeground);
+    }
+  }
+`;
+
+const Visibility = styled.span`
```

Comment:
I also introduce one of these in https://github.com/github/vscode-codeql/pull/1549. For these two PRs I wouldn't worry about it but once both are done I suggest we make a shared component for this.